### ISSUE TITLE
OpenCV job failure on `make opencv` (on 2.0-alpha21) - Works fine in 1.13.2

### ISSUE THREAD

**Issue by nadegepepin** [pachyderm/pachyderm#6339]
Went through openCV instructions in docs (https://docs.pachyderm.com/latest/getting_started/beginner_tutorial/) -> The final `pachctl list job` shows no error. See steps below

-`pachctl create repo images`
-`pachctl put file images@master:liberty.png -f http://imgur.com/46Q8nDz.png`
-`pachctl create pipeline -f https://raw.githubusercontent.com/pachyderm/pachyderm/master/examples/opencv/edges.json`
-`pachctl put file images@master:AT-AT.png -f http://imgur.com/8MN9Kg0.png
pachctl put file images@master:kitten.png -f http://imgur.com/g2QnNqa.png`
-`pachctl create pipeline -f https://raw.githubusercontent.com/pachyderm/pachyderm/master/examples/opencv/montage.json`
-`pachctl put file images@master -i images.txt`

**HOWEVER**
When running the example using the Makefile after a delete all (make opencv). 

opencv:
	pachctl create repo images
	pachctl create pipeline -f $(data_dir)/edges.json
	pachctl create pipeline -f $(data_dir)/montage.json
	pachctl put file images@master -i $(data_dir)/images.txt
	pachctl put file images@master -i $(data_dir)/images2.txt

the **first montage job** fails.

<img width="1075" alt="Screen Shot 2021-06-14 at 3 21 35 PM" src="https://user-images.githubusercontent.com/5490706/121947663-48dea680-cd24-11eb-9bc2-3cb685f71028.png">
<img width="669" alt="Screen Shot 2021-06-14 at 3 24 02 PM" src="https://user-images.githubusercontent.com/5490706/121947876-90fdc900-cd24-11eb-946a-6fd5e7946075.png">

Note: the same `make opencv` creates 4 jobs on 1.13.2 and 6 in alpha-21 - Race condition?

**Comment by Tryneus** [comment]
This is because the pipeline takes the root directory `/` as an input, but doesn't correctly handle when there are no images in the root directory.  The behavior has changed such that `/` is always returned when listing files, so the pipeline has a datum to process even if there is nothing in the directory.

**Comment by dgeorg42** [comment]
Why would the `montage` pipeline trigger if there were no images in the root directory, though? The only way I can think of would be if there was an empty commit, but that shouldn't be the case in this example.

**Comment by dgeorg42** [comment]
Never mind - I just saw the rest of the internal conversation on Slack, so I get it now. A pipeline is always run once when it is first created. When you run `make opencv` it first creates the repos, then it creates the two pipelines, each of which runs at creation time against the currently empty repos, and (for the reasons you stated) `montage` fails.  We don't see this when following the steps of the online tutorial because in that case you're not creating the `montage` pipeline until _after_ data has been added to the repos.
